src/defrag.c

```diff
@@ -280,6 +281,12 @@ robj *activeDefragStringOb(robj *ob) {
 static luaScript *activeDefragLuaScript(luaScript *script) {
     luaScript *ret = NULL;
 
+    /* In case we are in the process of eval some script we do not want to replace the script being run
+     * so we just bail out without really defragging here. */
+    if (scriptIsRunning()) {
+        return script;
+    }
+
```

Comment:
This is incorrect.  It should return NULL, per the comment on 278.

HOWEVER, I'd recommend moving this check into `defragLuaScripts`.  Then it will just skip spinning over the dictionary.